packages/near-fast-auth-signer-e2e-tests/utils/firebase.ts

```diff
@@ -12,25 +14,103 @@ import PageManager from '../pages/PageManager';
+  const randomWaitTime = getRandomWaitTime(1000, 5000);
+
+  await new Promise((resolve) => { setTimeout(resolve, randomWaitTime); });
+};
+
+export const isServiceAccountAvailable = () => {
+  if (
+    !serviceAccount.private_key_id
+    || !serviceAccount.private_key
+    || !serviceAccount.client_email
+    || !serviceAccount.client_id
+    || !serviceAccount.client_x509_cert_url
+  ) {
+    return false;
+  }
+  return true;
+};
+
 export const initializeAdmin = () => {
-  admin.initializeApp({
-    credential: admin.credential.cert(serviceAccount),
-  });
+  if (isServiceAccountAvailable() && !admin.apps.length) {
+    admin.initializeApp({
+      credential: admin.credential.cert(serviceAccount)
+    });
+  }
 };
 
 export const deleteAccount = async (userUid: string) => {
-  if (userUid) {
-    const user = await admin.auth().getUser(userUid);
+  try {
+    const user = userUid && await admin.auth().getUser(userUid);
     if (user.uid) {
+      // delete user auth record
       await admin.auth().deleteUser(userUid);
     }
+  } catch (error) {
+    if (error.code !== 'auth/user-not-found') {
+      console.error('Error deleting user auth record:', error);
+    }
+  }
+
+  try {
+    const db = admin.firestore();
+    const docRef = db.collection('users').doc(userUid);
+    // delete user collection record
+    await docRef.delete();
+  } catch (error) {
+    console.error('Error deleting user Firestore record:', error);
+  }
+};
+
+export const deleteUserByEmail = async (email: string) => {
+  try {
+    const user = await admin.auth().getUserByEmail(email);
+    if (user && user.uid) {
+      await deleteAccount(user.uid);
+    }
+  } catch (error) {
+    if (error.code !== 'auth/user-not-found') {
+      console.error('Error fetching user by email:', error);
+    }
   }
 };
 
-const addAccountPublicKeyToFirestore = async (accountId: string, publicKey: string) => {
-  const docRef = admin.firestore().collection('publicKeys').doc(publicKey);
+export const deletePublicKey = (publicKey: string) => {
+  const db = admin.firestore();
+  const docRef = db.collection('publicKeys').doc(publicKey);
+  return docRef.delete();
+};
 
-  await docRef.set({ accountId });
+export const addAccountToBeDeleted = async (account: MockAccount) => {
+  if (isServiceAccountAvailable() && admin.apps.length) {
+    const { accounts, ...rest } = JSON.parse(fs.readFileSync('testAccounts.json', 'utf-8'));
+    accounts.push(account);
+    fs.writeFileSync('testAccounts.json', JSON.stringify({ accounts, ...rest }, null, 2));
+  }
+};
+
+const addAccountIdPublicKey = async (publicKey: string, accountId: string) => {
+  if (isServiceAccountAvailable() && admin.apps.length) {
+    const db = admin.firestore();
+    const docRef = db.collection('publicKeys').doc(publicKey);
+    await docRef.set({ accountId });
+    const { publicKeys, ...rest } = JSON.parse(fs.readFileSync('testAccounts.json', 'utf-8'));
+    publicKeys.push({ publicKey, accountId });
```

Comment:
Same here, I think we need a locker. 

Maybe it's nice to introduce a utils function for I/O making sure we acquire the locker before update the file.